src/plugins/intel_cpu/src/nodes/executors/shl/shl_fullyconnected.cpp

```diff
@@ -84,9 +82,14 @@ ShlFCExecutor::ShlFCExecutor(const FCAttrs& attrs,
 }
 
 bool ShlFCExecutor::update(const MemoryArgs& memory) {
-    src.setShape(memory.at(ARG_SRC)->getDescPtr()->getShape().getStaticDims());
-    wei.setShape(memory.at(ARG_WEI)->getDescPtr()->getShape().getStaticDims());
-    dst.setShape(memory.at(ARG_DST)->getDescPtr()->getShape().getStaticDims());
+    auto update_shape = [&](ShlTensor& tensor, const VectorDims& shape) {
```

Comment:
Maybe add smt like "cloneWithNewDims" method to ShlTensor API? Same as for CPU MemoryDesc.
Otherwise we will need to copy-paste this part in each executor.

Reply:
Discussed offline: implemented new method `ShlTensor::cloneWithNewShape(...)` in https://github.com/openvinotoolkit/openvino/pull/23964/commits/a3a1003ac7fef92c12a35cbcc943c1f7ec3d5f22